libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -378,7 +380,11 @@ public void updateCameraRouteOverview() {
    * @param options with containing route / coordinate data
    */
   public void startNavigation(NavigationViewOptions options) {
-    initializeNavigation(options);
+    try {
+      initializeNavigation(options);
```

Comment:
i wonder if we're just fixing a symptom here, any idea why NPE is happening here?

Reply:
The issue here is race condition. The use case is as follows:
Click on start navigation: 
It invokes `NavigationView/startNavigation/initializeNavigation/initializeNavigationListeners`
`initializeNavigationListeners` makes a call to `navigationMap.addListeners()`
Meanwhile all this is happening, the user destroys the Activity/Fragment thereby resulting in the `onDestroy` being called. As the `onDestroy` is called, all member variables become null. The call to this line `navigationMap.addListeners()` now throws a null pointer exception as `navigationMap` is now null. 

So, the issue here is that meanwhile the app is initializing the variables and starting to navigate, if the user happens to destroy the app completely, any method calls that depends on any of the member variables will throw a null pointer exception.

Easiest fix to this issue is to add null check everywhere, which I don't like doing. So, I took this approach of adding a try catch. Apparently this is a much rare use case and it is quite a good amount of refactor required to prevent this from happening.

We decided, that this ticket could be resolved by asking the user to wrap their call to `startNavigation` inside try catch meanwhile we open another ticket to refactor our `NavigationView` and `NavigationViewModel` 

@andrlee Let me know your thoughts on that

Reply:
@abhishek1508 let's put the fix in and cut the ticket with the description of the issue and tag it with `1.0.0` milestone.

Reply:
> We decided, that this ticket could be resolved by asking the user to wrap their call to `startNavigation` inside try catch meanwhile we open another ticket to refactor our `NavigationView` and `NavigationViewModel`

If that's the case, shouldn't we remove the `try catch` altogether? Main problem with `try catch` here is that we could be hiding any potential `NullPointerException`s coming from methods down the stack in a normal scenario when running `startNavigation`. That could leave the navigation session in an undesired state (causing confusion to end-users) instead of crashing the app.

Reply:
@Guardiola31337 would love to learn about suggestions how NPE can be fixed more efficiently

Reply:
> If that's the case, shouldn't we remove the `try catch` altogether? 
>>
Yeah that is what I was saying is that I could close this PR and inform the reporter of the issue of the workaround while we work on refactoring `NavigationView` and `NavigationViewModel`

Reply:
Although I'm not a big fan of adding `null` checks everywhere either this is what has been done in the past to avoid / prevent these kind of issues. That being said, the `NavigationView` / `NavigationViewModel` setup is been forever and definitely needs a bigger refactor with special focus on the lifecycle management.

So if we don't want to do 👇 
> Yeah that is what I was saying is that I could close this PR and inform the reporter of the issue of the workaround

maybe we could go ahead and add the `null` checks (for now) and work on the above-mentioned refactor in a separate PR targeting `1.0.0` milestone. What do you think?

Reply:
Based on our discussion, we are not going to fix this either with try/catch and null check.
**Null Check** is resulting in NPE at other places.
**try/catch** would catch all the exceptions thereby making it difficult for us to debug further issues.

The only solution to this bug is to revisit our `NavigationView`/`NavigationViewModel` architecture.